### DISCUSSION TITLE
prisma client typing question

### DISCUSSION THREAD

**Discussion by bitsofinfo** [prisma/prisma#15261] [Q&A]
Hi! new to prisma and trying to understand some basics, trying to do something simple (or seems should be simple) but running into some typing crazyness.

```

    async update(updatedBy:Principal, data: Prisma.BlogUpdateInput): Promise<BlogEntry> {
        data.updatedBy = updateBy.principalId;
        return await this.prisma.blogEntry.update({
                                where: {
                                    id: data.id
                                },
                                data: data
                            });
    }
```

but this yields me an syntax error stating:

```
Type 'string | StringFieldUpdateOperationsInput' is not assignable to type 'string'.
  Type 'StringFieldUpdateOperationsInput' is not assignable to type 'string'.ts(2322)
index.d.ts(4311, 5): The expected type comes from property 'id' which is declared here on type 'BlogEntryWhereUniqueInput'
```

BlogUpdateInput.id is just a string... what "generated" type should I be using to on my update() method that wraps the call to prisma? This is a bit confusing (edited)

**Comment by austincrim** [comment]
Hey @bitsofinfo 👋,

Could you share your Prisma schema?

**Reply by vladi-stevanovic** [reply]
👋 Hello @bitsofinfo is this still an issue for you? Let us know if you can share your schema or if you've already solved it, we'd love to know how! It would be super useful for other users facing this problem!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
